charts/retool/templates/telemetry_deployment.yaml

```diff
@@ -0,0 +1,158 @@
+{{- if .Values.telemetry.enabled }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "retool.fullname" . }}-telemetry
+  labels:
+    {{- include "retool.telemetry.labels" . | nindent 4 }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "retool.telemetry.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "retool.telemetry.selectorLabels" . | nindent 8 }}
+      annotations:
+        {{- if or .Values.telemetry.customVectorConfig .Values.telemetry.customGrafanaAgentConfig }}
+        checksum/custom-config: {{ include (print $.Template.BasePath "/telemetry_configmap.yaml") . | sha256sum | quote }}
+        {{- end }}
+    spec:
+      serviceAccountName: {{ include "retool.telemetry.serviceAccountName" . }}
+      containers:
+      - name: telemetry
+        image: '{{ .Values.telemetry.image.repository }}:{{ .Values.telemetry.image.tag | default .Values.image.tag }}'
+        imagePullPolicy: {{ .Values.telemetry.image.pullPolicy }}
+        command: ['retool-telemetry']
+        resources:
+          {{ toYaml .Values.telemetry.resources | nindent 10 }}
+        env:
+          - name: VECTOR_SELF_NODE_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: spec.nodeName
+          - name: VECTOR_SELF_POD_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.name
+          - name: VECTOR_SELF_POD_NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace
+          - name: PROCFS_ROOT
+            value: "/host/proc"
+          - name: SYSFS_ROOT
+            value: "/host/sys"
+          {{- if and (not .Values.externalSecrets.enabled) (not .Values.externalSecrets.externalSecretsOperator.enabled) }}
+          - name: LICENSE_KEY
+            valueFrom:
+              secretKeyRef:
+                {{- if .Values.config.licenseKeySecretName }}
+                name: {{ .Values.config.licenseKeySecretName }}
+                key: {{ .Values.config.licenseKeySecretKey | default "license-key" }}
+                {{- else }}
+                name: {{ template "retool.fullname" . }}
+                key: license-key
+                {{- end }}
+          {{- end }}
+          - name: RTEL_DEPLOYMENT_MODE
+            value: 'kubernetes-helm'
+          - name: RTEL_HELM_RELEASE_NAME
+            value: {{ .Release.Name | quote }}
```

Comment:
maybe: align this to https://github.com/tryretool/retool_development/blob/dev/backend/src/server/modules/jobsRunner/startupTelemetry.ts#L49-L62

DEPLOYMENT_TEMPLATE_TYPE
DEPLOYMENT_TEMPLATE_VERSION

Reply:
I think I will need to keep the env vars `RTEL_DEPLOYMENT_MODE` and `DEPLOYMENT_TEMPLATE_TYPE` separate after all, because I think they will diverge for other deployment templates. I'll still at least unify the values so e.g. `k8s-helm` is used by both, and also add `DEPLOYMENT_TEMPLATE_TYPE` as tags to metrics/logs when present.

Reply:
ack ok 👍